provisioning_templates/snippet/redhat_register.erb

```diff
@@ -3,90 +3,166 @@ kind: snippet
-#                                             for example:
-#                                             http://fqdn/pub
-#                                             /katello-ca-consumer-latest.noarch.rpm)
+#   subscription_manager_password = <password>  Password for subscription-manager
 #
-#   subscription_manager_host = <hostname> (deprecated for
-#                                           subscription_manager_certpkg_url:
-#                                           hostname of SAM/Katello
-#                                           installation, if using SAM.
-#                                           hostname is used to determine the
-#                                           consumer cert url.)
+#   subscription_manager_certpkg_url = <url>    Custom certificate package URL
 #
-#   subscription_manager_org = <org name> (organization name, if using
-#                                          SAM/Katello)
+#   subscription_manager_atomic_url = <url>     Script used for configuring Atomic servers
+#                                               for subscription-manager
 #
-#   subscription_manager_repos = <repos> (comma separated list of repos (like
-#                                         rhel-6-server-optional-rpms) to
-#                                         enable after registration)
+#   subscription_manager_org = <org name>       Organization name, if required
 #
-#   subscription_manager_pool = <pool> (specific pool to be used for
-#                                       registration)
+#   subscription_manager_repos = <repos>        Additional repositories to enable
+#                                               after registration
 #
-#   http-proxy = <host> (proxy hostname to be used for registration)
+#   subscription_manager_pool = <pool>          Specific subscription pool to use
 #
-#   http-proxy-port = <port> (proxy port to be used for registration)
+#   http-proxy = <host>                         Proxy hostname to be used for registration
 #
-#   http-proxy-user = <user> (proxy user to be used for registration)
+#   http-proxy-port = <port>                    Proxy port to be used for registration
 #
-#   http-proxy-password = <password> (proxy password to be
-#                                           used for registration)
+#   http-proxy-user = <user>                    Proxy user to be used for registration
 #
+#   http-proxy-password = <password>            Proxy password to be used for registration
 #
-# Set this parameter regardless of which registration method you're using:
+# Set these parameters if you're using rhnreg_ks:
 #
-#   activation_key = <key>      (activation key string, not needed if using
-#                                subscription-manager with hosted RHN)
+#   spacewalk_host = <hostname>                 Hostname of Spacewalk server
 #
+%>
+
+<%
+  # Katello or subscription-manager:
+  if host_param_true?('subscription_manager') || host_param('kt_activation_keys')
```

Comment:
@lzap We detect subscription-manager for katello with `kt_activation_keys` host parameter.

Reply:
(which katello adds when you have an activation key selected)